pkg/manifests/manifests.go

```diff
@@ -2558,6 +2569,72 @@ func (f *Factory) NewValidatingWebhook(manifest io.Reader) (*admissionv1.Validat
 	return NewValidatingWebhook(manifest)
 }
 
+func (f *Factory) NewConsolePlugin(manifest io.Reader) (*consolev1.ConsolePlugin, error) {
+	return NewConsolePlugin(manifest)
+}
+
+func (f *Factory) MonitoringPlugin() (*consolev1.ConsolePlugin, error) {
+	return f.NewConsolePlugin(f.assets.MustNewAssetReader(MonitoringPlugin))
+}
+
+func (f *Factory) MonitoringPluginDeployment() (*appsv1.Deployment, error) {
+	d, err := f.NewDeployment(f.assets.MustNewAssetReader(MonitoringPluginDeployment))
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure console-plugin container is present even if config isn't defined so that,
+	// we validate that deployment has the expected container name. Thereby avoiding
+	// any suprises should user add config later.
```

Comment:
not sure what you mean by "any surprises should user add config later"?

Reply:
The initial code I wrote was to return if the config for console-plugin is nil. 
Now .. lets say the `console-plugin` container is absent (for whatever reason .. may be a typo, jsonnet issue). User then adds the config to set resource limits. They would then start experiencing error which may be confusing (like it confused me).
By ensuring that the container is present, errors will be caught earlier.

Reply:
got it! the other option would be to fail hard (e.g. panic) since it would mean that the asset and the Go code are out of sync.